spark/src/main/scala/io/delta/tables/DeltaMergeBuilder.scala

```diff
@@ -155,6 +157,8 @@ class DeltaMergeBuilder private(
   with Logging
   {
 
+  private var schemaEvolutionEnabled: Boolean = false
```

Comment:
I did that before with a default value of `false`, but I still can't pass the compatibility test.
I also tried to create a `apply()` method without the `schemaEvolutionEnabled` parameter in the companion object, but also failed to pass the test.

I'll find a Scala expert and ask.

Reply:
I didn't run into this issue when adding WHEN NOT MATCHED BY SOURCE because all clauses are passed as a single argument and I didn't need to add one.
I think the current approach is ok, I was mainly curious whether this was to keep the API stable

Reply:
ABI fixed! [`209f975` (#2737)](https://github.com/delta-io/delta/pull/2737/commits/209f9752839e5f43681ea0511939d35d2dc5d75f)